F21/rjuliang/LowerCaser/assets/process.js

```diff
@@ -0,0 +1,136 @@
+                lowerCasedWord = resolveIrishWord(word, language);
+            }
+
+            
+
+
+            //Then, we check if a user is triggering the function and not a unit test (unitTest is False)
+            if(unitTest == false) {
+
+                //Then, if it is not a test, we populate the content of the lowercasedDiv with the result                 
+                document.getElementById('lowercasedDiv').style.display = "block";
+                document.getElementById('lowercased').innerHTML="<p>"+lowerCasedWord+"</p>";
+
+            }
+            
+            //Finally, we return the value of the lowercased word
+            return lowerCasedWord;
+
+        } else {
+
+            //If the user did not provide a word, we show an alert
+            alert('Type a word to lower case');
+
+            return false;
+        }
+        
+    }
+
+    function resolveIrishWord (word, language){
+        //First, we define the variables
+        let resolvedWord; // Generic variable where we will store the final lowercased word
+        let firstCharacter = word.charAt(0);//Gets the FIRST character of the word
+        let secondCharacter = word.charAt(1);//Gets the SECOND character of the word
+
+        if(firstCharacter != "n" && firstCharacter != "t"){
+
+            //If the first character is NOT "n" or "t", we run the default native toLocaleLowerCase
+            resolvedWord = word.toLocaleLowerCase(language.toString());
+
+        } else {
+
+            //If the first character is "n" or "t", we run our customization by triggering the isVowel function with the second character (see function in line 116)
+            if(isVowel(secondCharacter) == false){
+
+                //If the second character is NOT an uppercased vowel, we run the default native toLocaleLowerCase
+                resolvedWord = word.toLocaleLowerCase(language.toString());
+            } else {
+
+                //If the character IS an uppercased vowel
+
+                //We convert the word into an array
+                const arrayOfWord = word.split("");
+
+                //Then, we insert the dash (-) in the second position of the array
+                arrayOfWord.splice(1,0,"-");
+
+                //Join the word into a string and converting the string to a lowercase
+                resolvedWord = arrayOfWord.join("").toLowerCase();                   
+            }
+        }
+
+        //Finally, we return the resolvedWord
+        return resolvedWord;
+    }
+
+    function isVowel(character) {
+        /**
+         * In here, we return a boolean (TRUE | FALSE) if the character is equal to one of the uppercased vowels:
+         * (A,E,I,O,U,Á,É,Í,Ó,Ú)
+         */
+        let result;
+
+        result = character === "A" || character === "E" || character === "I" || character === "O" || character === "U" || character === "Á" || character === "É" || character === "Í" || character === "Ó" || character === "Ú";
+
```

Comment:
There is an option to use array list here. That would be easier than defining each character.

Reply:
You are right, I will make the updates to the code with an array for this function